hugegraph-rocksdb/src/main/java/com/baidu/hugegraph/backend/store/rocksdb/RocksDBStore.java

```diff
@@ -634,88 +632,45 @@ protected Session session(HugeType tableType) {
-                             snapshotPath);
-                LOG.debug("The origin data path: {}", originDataPath);
-                LOG.debug("The snapshot data path: {}", snapshotPath);
+                RocksDBSessions sessions = entry.getValue();
+                Path snapshotPath = sessions.buildSnapshotPath(originDataPath,
+                                                               snapshotPrefix,
+                                                               deleteSnapshot);
+                snapshotPaths.put(snapshotPath, sessions);
+            }
 
+            for (Map.Entry<Path, RocksDBSessions> entry :
+                 snapshotPaths.entrySet()) {
+                Path snapshotPath = entry.getKey();
                 RocksDBSessions sessions = entry.getValue();
-                RocksDB rocksdb = sessions.createSnapshotRocksDB(
-                                  snapshotPath.toString());
-                Path snapshotLinkPath = Paths.get(originDataPath + "_link");
-                createCheckpoint(rocksdb, snapshotLinkPath.toString());
-                rocksdb.close();
+                sessions.resumeSnapshot(snapshotPath.toString());
 
-                fileNameMaps.put(originDataPath, snapshotLinkPath);
-                uniqueParents.add(snapshotPath.getParent().toString());
-            }
-            E.checkState(!fileNameMaps.isEmpty(),
-                         "The file name maps shouldn't be empty");
-            /*
-             * NOTE: must close rocksdb instance before deleting file directory,
-             * if close after copying the snapshot directory to origin position,
-             * it may produce dirty data.
-             */
-            for (RocksDBSessions sessions : this.sessions()) {
-                sessions.forceCloseRocksDB();
-            }
-            // Move snapshot file to origin data path
-            for (Map.Entry<Path, Path> entry : fileNameMaps.entrySet()) {
-                File originDataDir = entry.getKey().toFile();
-                File snapshotLinkDir = entry.getValue().toFile();
-                try {
-                    if (originDataDir.exists()) {
-                        LOG.info("Delete origin data directory {}",
-                                 originDataDir);
-                        FileUtils.deleteDirectory(originDataDir);
-                    }
-                    FileUtils.moveDirectory(snapshotLinkDir, originDataDir);
-                    LOG.info("Move snapshot directory {} to {}",
-                             snapshotLinkDir, originDataDir);
-                } catch (Exception e) {
-                    throw new BackendException("Failed to move %s to %s", e,
-                                               snapshotLinkDir, originDataDir);
+                Path parentPath = snapshotPath.getParent();
+                if (Files.list(parentPath).count() == 0) {
+                    FileUtils.deleteDirectory(parentPath.toFile());
```

Comment:
seems still not use move directory if needed to delete snapshot

Reply:
I didn't understand.